contracts/contracts/ProviderRegistry.sol

```diff
@@ -194,53 +189,37 @@ contract ProviderRegistry is
         require(successFee, "fee recipient transfer failed");
     }
 
-    /**
-     * @dev Withdraw funds to the bidder.
-     * @param bidder The address of the bidder.
-     */
-    function withdrawBidderAmount(address bidder) external nonReentrant {
-        require(bidderAmount[bidder] > 0, "Bidder Amount is zero");
-
-        bidderAmount[bidder] = 0;
-
-        (bool success, ) = bidder.call{value: bidderAmount[bidder]}("");
-        require(success, "Couldn't transfer to bidder");
+    /// @dev Requests withdrawal of the staked amount.
+    function requestWithdrawal() external {
+        require(providerStakes[msg.sender] > 0, "No stake to withdraw");
+        withdrawalRequests[msg.sender] = block.timestamp;
```

Comment:
This function shouldn't be callable once it's already been called for the sender 
```suggestion
        require(withdrawalRequests[msg.sender] == 0);
        withdrawalRequests[msg.sender] = block.timestamp;
```